cluster/gce/gci/configure-helper.sh

```diff
@@ -1706,17 +1706,41 @@ function prepare-etcd-manifest {
   if [[ -n "${INITIAL_ETCD_CLUSTER_STATE:-}" ]]; then
     cluster_state="${INITIAL_ETCD_CLUSTER_STATE}"
   fi
+
+  # Configure mTLS for etcd peers.
   if [[ -n "${ETCD_CA_CERT:-}" && -n "${ETCD_PEER_KEY:-}" && -n "${ETCD_PEER_CERT:-}" ]]; then
     etcd_creds=" --peer-trusted-ca-file /etc/srv/kubernetes/etcd-ca.crt --peer-cert-file /etc/srv/kubernetes/etcd-peer.crt --peer-key-file /etc/srv/kubernetes/etcd-peer.key -peer-client-cert-auth "
     etcd_protocol="https"
   fi
 
-  # mTLS should only be enabled for etcd server but not etcd-events. if $1 suffix is empty, it's etcd server.
+  local host_ip=""
+  local etcd_additional_listen_client_urls=""
+  local etcd_additional_listen_metrics_urls=""
+  if [[ ${ETCD_LISTEN_ON_HOST_IP:-} == "true" ]]; then
+    # Override host IP if provided.
+    if [[ -n ${HOST_PRIMARY_IP:-} ]]; then
+      host_ip=${HOST_PRIMARY_IP}
+    else 
+      host_ip=$(hostname -i)
+    fi
```

Comment:
I have a significant concerns about this for existing GKE clusters.
Did you check the flow of:
- creating GKE *regional* clusters using the old flow
- try to upgrade them to the flow that involved this logic

Reply:
I ran a test on that.
Initially created a regional cluster with version 1.16.13-gke.401
Upgraded to my custom build 1.19.2-gke.2200.2+0f0e25619b0efe
Both master and node upgrade were successful.

Reply:
I think we need to make sure that the automation tests for the upgrade scenario pass in this case. You might be blocked by issue #94487.

Reply:
Ok let me run e2e upgrade tests and see if it works. Shihang seemed to have merged a few fixes already.